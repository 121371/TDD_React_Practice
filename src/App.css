.App {
  text-align: center;
  padding-left: 2.5rem;
  padding-right: 2.5rem;
  min-height: 100vh;
  font-size: calc(10px + 2vmin);
  color: white;
  background-color: rgb(82, 78, 78);
  position: relative;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  height: 40px;
  background-color: black;
  width:100%;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.App-footer {
  position: absolute;
  bottom: 0;
  text-align: center;
  background-color: black;
  width: calc(100% - 5rem);
  padding-top: 20px;
  padding-bottom: 20px;
}
