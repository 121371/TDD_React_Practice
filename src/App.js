import Login from "./components/Login/Login";

function App() {
  return (
    <div className="App">
      <header className="App-header">This is a login page</header>
      <Login />
    </div>
  );
}

export default App;
