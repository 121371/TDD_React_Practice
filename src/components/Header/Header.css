.header {
    display: flex;
    background-color: black;
    color: white;
    display: flex;
    justify-content: space-between;
    margin-bottom: 24px;
}

.header-text {
    text-align: center;
    width: 100%;
}

.nav-links ul {
  display: flex;
}

.nav-links ul li {
    padding-right: 10px;
    list-style-type: none;
}

.nav-links a {
    text-decoration: none;
}