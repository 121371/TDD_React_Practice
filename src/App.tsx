import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from './components/Header/Header';
import Login from './components/Login/Login';
import Register from './components/Register/Register';

import './App.css';

function App(): React.ReactNode {
  return (
    <div className="App">
      <header>
        <Header />
      </header>
      <Switch>
        <Route path="/login" component={Login} />
        <Route path="/register"  component={Register} />
      </Switch>
      <footer className="App-footer">This is Footer</footer>
    </div>
  );
}

export default App;
