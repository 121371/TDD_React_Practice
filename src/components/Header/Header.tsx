import React from "react";
import { Link } from "react-router-dom";

import "./Header.css";

const Header: React.FC = () => {
  return (
    <div className="header">
      <div className="header-text">
        <p>Welcome to our app</p>
      </div>
      <div className="nav-links">
        <ul>
          <li>
            <Link to="/login">Login</Link>
          </li>
          <li>
            <Link to="/register">Register</Link>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Header;
