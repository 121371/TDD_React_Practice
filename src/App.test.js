import { shallow } from "enzyme";
import { Route } from 'react-router-dom';

import Header from "./components/Header/Header";
import App from "./App";

describe("app component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('App component renders correctly', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("child component named Header exists", () => {
    expect(wrapper.find(Header).exists()).toBeTruthy();
  });

  describe('App routing', () => {
    let routeWrapper;
    beforeEach(() => {
      routeWrapper =  wrapper.find(Route);
    });
    it("Two routes should be available", () => {
      
      expect(routeWrapper.length).toBe(2);
     });

     it("route for login component", () => {
      expect(routeWrapper.first().props().path).toBe('/login');
     });

     it("route for register component", () => {
      expect(routeWrapper.last().props().path).toBe('/register');
     });
  });

  it('Footer should be present with same text', () => {
    expect(wrapper.find('footer')).toBeTruthy();
    expect(wrapper.find('footer').text()).toBe('This is Footer');
  });
});
