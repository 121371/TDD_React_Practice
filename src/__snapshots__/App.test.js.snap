// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component App component renders correctly 1`] = `
Array [
  <div
    className="App"
  >
    <header>
      <Header />
    </header>
    <Switch>
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/register"
      />
    </Switch>
    <footer
      className="App-footer"
    >
      This is Footer
    </footer>
  </div>,
]
`;
